### PR TITLE
Enable Dark mode theme

### LINKED ISSUE #739: Dark Mode for Web Components
### Theme Setting

Implement 'dark' and 'light' theme setting for the web components
*Maybe* add a 'preference' mode which will automatically apply the user's preferred theme? This could be its own story outside of the Barebones requirements

There should be two ways to set the theme:
- [x] Page Level - Set the theme on the page's body with a class that the components can read from. This would apply the theme globally
- [x] Component Level - Set the theme directly on the component with a CSS attribute, which would override the Page Level setter

---

### 

- [x] #787
- [x] #788 
- [x] #789 
- [x] #796
- [x] #797
- [x] #800
- [x] #801
- [x] #802
- [x] #803
- [x] #804
- [x] #805
- [ ] #843
- [x] #853
- [x] #874
- [ ] maybe #847

Once we've got our pattern for setting the theme on the Page Level, a `ThemeToggle` component should be created that will switch the theme globally. This will allow consumers of the library to get 'out of the box' theme toggling

### PR DESCRIPTION
**Pull request by msankaran0712**
Fixes #739 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update

## How Has This Been Tested?

Browser

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (including CHANGELOG updates)
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

**Review by egunther39 — DISMISSED** [review]
Nice work!

**Review by ryan-henness-trimble — DISMISSED** [review]

**Comment by coliff** [comment]
I don't think this is a blocking issue but the dark mode toggle is sometimes working inconsistently. 

I've noticed this issue first on my iPad with latest iPadOS 16.1 but also with Edge v107 on Windows 10. Screenshot here...
image

File: stencil-workspace/storybook/.storybook/preview-body.html [hunk 1]
```diff
@@ -3,3 +3,19 @@
     font-family: "Open Sans", sans-serif;
   }
 </style>
+<script>
```
Comment (msankaran0712) [inline stencil-workspace/storybook/.storybook/preview-body.html:6]: @ryan-henness-trimble @coliff this is a workaround for an issue reported by Christian which is about the Canvas tab not being consistent with the theme toggler state. I could not find a better way to fix it so If you find any issue with this approach please let me know.

Comment (msankaran0712) [reply stencil-workspace/storybook/.storybook/preview-body.html:6]: The script will listen to any change done to the `body` element and the theme toggler tool adds a class for each theme so the script uses it to find the theme selected.

Comment (msankaran0712) [reply stencil-workspace/storybook/.storybook/preview-body.html]: that fix broke the workflow so reverted it.

**Review by ryan-henness-trimble — APPROVED** [review]

**Pull request merged**